### DISCUSSION TITLE
Is it possible to "disable" polling for one expose that is not used?

### DISCUSSION THREAD

**Discussion by pauloon** [Koenkk/zigbee2mqtt#14601] [Devices]
Hello all,

I have a Tuya presence sensor (mmWave - TS0601 - _TZE200_ztc6ggyl) that works perfectly with Z2M, but there is one expose called "target_distance" which is kind of useless for me, but the device keeps polling this every 0.5 second, which creates a lot of unnecessary traffic in my Zigbee network, specially in my case, where I have many of these installed.

![image](https://user-images.githubusercontent.com/48531854/197366641-4b377f7a-62bb-42a5-81e9-f3f0712969ff.png)
 
Is it possible to do something to "disable" this expose from being updated, or ignore it?
Maybe there's a way of sending a command at Dev Console to make this happen.

![image](https://user-images.githubusercontent.com/48531854/197366760-22b08fc9-d9cc-488f-b92b-4eb9bfddc061.png)

Here's the detail about the device:
![image](https://user-images.githubusercontent.com/48531854/197366875-1bf3003b-1986-4058-88fe-59f8910eaf9f.png)

Please, help!

Thanks,
Paulo.

**Comment by pauloon** [comment]
Plus: I already checked in "settings" and there is no configuration available to set polling interval or disable it....

**Reply by budcalloway** [reply]
hi Pauloon, you may try the filtering or debounce options in the settings of your device
![test](https://user-images.githubusercontent.com/83547288/197378724-9dae913c-d02a-4f97-a635-21d19b92c99d.PNG)
I used the debounce option to reduce the amount of data sent on the MQTT chanel. Never used the filters yet

**Reply by pauloon** [reply]
Thanks for the tip. I already set a debounce and filter and noticed that indeed reduced amount of data published to MQTT.
But I think this will not work reducing the Zig network traffic, unfortunatelly.

Thanks,
Paulo.

**Comment by Koenkk** [comment]
You cannot. The TuYa device just send these values (push, not poll!) and there is no way to disable this (since TuYa uses non-standard Zigbee)

**Reply by pauloon** [reply]
Ok. Thank you! Will see how to handle this.

**Comment by tobiasbrage** [comment]
I'm using the exact same sensor and tried to find a way to disable the target_distance expose as i have three sensors now and each sensor is updating the target_distance every second or so. I don't even need to know the distance so it's just unecessary traffic. If someone finds a way to disable it please let me know!

If TuYa indeed is using non-standard Zigbee I might replace the sensors to some Aqara ones which dosen't spam my Zigbee network with useless information.

**Comment by pauloon** [comment]
Dear @tobiasbrage ...
As Koenkk said, it is not possible to disable this.
To light up your Z2M network you can use these sensors with a Tuya Zigbee Hub, integrating with HA via Tuya Cloud.
Or you can move to other mmWave sensors like Aqara FP1, which performs a little different.
I did this and I'm satisfied.

Thanks,
Paulo.

**Reply by ajunio3** [reply]
> Dear @tobiasbrage ... As Koenkk said, it is not possible to disable this. To light up your Z2M network you can use these sensors with a Tuya Zigbee Hub, integrating with HA via Tuya Cloud. Or you can move to other mmWave sensors like Aqara FP1, which performs a little different. I did this and I'm satisfied.
> 
> Thanks, Paulo.

Hello there

I'm in the same situation. I've bought one sensor for testing and as long as it detects a presence it keeps pushing mqtt messages every second or so to measure target distance
But...the plan is to use 15 of this devices in the sane network. Doyou think it's gona be a problem with traffic? The entire network would have 90 devices or so

I've also tested aqara Fp1 and it doe's not perforn this way. But the ceiling mount design of this one fit's perfectly my needs

**Reply by markhaines** [reply]
I'm in exactly the same situation - really like the form factor but don't want to kill my network. Did you resolve this in the end?

**Comment by marcog20** [comment]
Hello to all! I'm also interesting on this topic, since I bought a couple of them to use instead of classic motion detection PIR sensor.
The device is working very well, the issue is about the huge traffic on the zigbee network that often causes a crash of Z2M addon.
Anyone solved this?

**Reply by pauloon** [reply]
I solved by changing my sensors to Aqara FP1....
There is also a possibility if you use them with a Tuya Hub via the cloud integration with HA.

**Comment by capyvara** [comment]
This is old but by the way on the Tuya app the sensor seems to have a "report distance" configuration that can be disabled, manual says it's for debugging only as it generate traffic, z2m has a "Find switch" to control it.
